harness/assert.js

```diff
@@ -62,3 +62,64 @@ assert.throws = function (expectedErrorConstructor, func) {
 
     $ERROR('Expected a ' + expectedErrorConstructor.name + ' to be thrown but no exception was thrown at all');
 };
+
+(function() {
+  function classOf(object) {
+    // Argument must not be null or undefined.
+    var string = Object.prototype.toString.call(object);
+    // String has format [object <ClassName>].
+    return string.substring(8, string.length - 1);
+  }
+
+  function deepObjectEquals(a, b) {
+    var aProps = Object.keys(a);
```

Comment:
Hmm yeah, good call I think. Object.keys is an ES5 method we shouldn't depend on unless absolutely necessary. Doesn't seem necessary here, as gross as for-in enumeration is.